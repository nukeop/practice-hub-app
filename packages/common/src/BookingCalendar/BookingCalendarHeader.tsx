import { format } from 'date-fns';
import { ChevronLeft, ChevronRight } from 'lucide-react';
import { FC } from 'react';

import { Button } from '../Button';

type BookingCalendarHeaderProps = {
  onPrevious: () => void;
  onNext: () => void;
  onToday: () => void;
  weekDates: Date[];
};

export const BookingCalendarHeader: FC<BookingCalendarHeaderProps> = ({
  onPrevious,
  onNext,
  onToday,
  weekDates,
}) => {
  return (
    <div className="flex items-center justify-between border-b border-stone-800 px-2 py-1">
      <h2 className="bg-gradient-to-b from-stone-200 to-stone-300 bg-clip-text text-lg font-bold text-transparent">
        {format(weekDates[0], 'd')} â€” {format(weekDates[6], 'd MMM yyyy')}
      </h2>
      <div className="flex gap-2">
        <Button variant="outline" size="sm" onClick={onToday}>
          Today
        </Button>
        <Button
          data-testid="prev-week"
          variant="outline"
          size="icon"
          onClick={onPrevious}
        >
          <ChevronLeft className="h-4 w-4" />
        </Button>
        <Button
          data-testid="next-week"
          variant="outline"
          size="icon"
          onClick={onNext}
        >
          <ChevronRight className="h-4 w-4" />
        </Button>
      </div>
    </div>
  );
};
