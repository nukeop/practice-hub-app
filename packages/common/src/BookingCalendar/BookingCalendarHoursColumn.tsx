import { format } from 'date-fns';

export const BookingCalendarHoursColumn = () => {
  const hours = Array.from({ length: 24 }, (_, i) => i);
  return (
    <div className="sticky left-0 z-10 h-full bg-transparent">
      <div className="h-4 border-none border-stone-800" />
      <div className="h-4 border-b border-stone-800" />
      {hours.map((hour) => (
        <div
          key={hour}
          className="flex h-4 items-center border-b border-r border-stone-800 px-2 py-2 text-sm text-stone-400"
        >
          {format(new Date().setHours(hour, 0), 'HH:mm')}
        </div>
      ))}
    </div>
  );
};
