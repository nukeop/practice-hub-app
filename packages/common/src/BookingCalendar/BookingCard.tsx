import { cva, VariantProps } from 'class-variance-authority';
import { format } from 'date-fns';
import { FC } from 'react';

const bookingVariants = cva(
  [
    'absolute left-0 right-0 mx-1 flex items-center overflow-hidden rounded-md px-2 text-xs font-medium ring-1 text-stone-200',
  ],
  {
    variants: {
      color: {
        blue: 'bg-blue-900 text-blue-100 ring-blue-500',
        red: 'bg-red-900 text-red-100 ring-red-500',
        green: 'bg-green-900 text-green-100 ring-green-500',
        yellow: 'bg-yellow-900 text-yellow-100 ring-yellow-500',
      },
    },
    defaultVariants: {
      color: 'blue',
    },
  },
);

type BookingCardProps = {
  start: Date;
  end: Date;
  top: number;
  height: number;
} & VariantProps<typeof bookingVariants>;

export const BookingCard: FC<BookingCardProps> = ({
  start,
  end,
  top,
  height,
  color,
}) => {
  return (
    <div
      data-testid="booking"
      style={{
        top: `${top}px`,
        height: `${height}px`,
      }}
      className={bookingVariants({ color })}
    >
      <div className="relative w-auto max-w-full truncate">
        {format(start, 'HH:mm')} - {format(end, 'HH:mm')}
      </div>
    </div>
  );
};
