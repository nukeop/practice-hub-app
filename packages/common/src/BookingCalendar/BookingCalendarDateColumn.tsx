import { differenceInMinutes, format } from 'date-fns';
import { FC } from 'react';

import { Booking } from './BookingCalendar';
import { BookingCard } from './BookingCard';

type BookingCalendarDateColumnProps = {
  date: Date;
  bookings: Booking[];
};

export const BookingCalendarDateColumn: FC<BookingCalendarDateColumnProps> = ({
  date,
  bookings,
}) => {
  const hours = Array.from({ length: 24 }, (_, i) => i);
  return (
    <div className="relative h-full">
      <div className="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium">
        <div className="flex flex-row gap-2 text-center">
          <div className="text-base">{format(date, 'EEE')}</div>
          <div className="text-base">{format(date, 'd.M')}</div>
        </div>
      </div>
      <div className="relative">
        {hours.map((hour) => (
          <div
            key={hour}
            className="group relative h-4 border-b border-r border-stone-800 py-2"
          />
        ))}
        {bookings.map((booking) => {
          const startOfDay = new Date(booking.start).setHours(0, 0, 0, 0);
          const top =
            (differenceInMinutes(booking.start, startOfDay) / 60) * 17;
          const height =
            (differenceInMinutes(booking.end, booking.start) / 60) * 17;

          return (
            <BookingCard
              key={booking.id}
              start={booking.start}
              end={booking.end}
              top={top}
              height={height}
              color={booking.color}
            />
          );
        })}
      </div>
    </div>
  );
};
