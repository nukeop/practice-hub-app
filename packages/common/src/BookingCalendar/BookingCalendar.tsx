'use client';

import {
  addDays,
  differenceInMinutes,
  format,
  isSameDay,
  startOfWeek,
} from 'date-fns';
import { useState } from 'react';

import { BookingCalendarDateColumn } from './BookingCalendarDateColumn';
import { BookingCalendarHeader } from './BookingCalendarHeader';
import { BookingCalendarHoursColumn } from './BookingCalendarHoursColumn';

export type Booking = {
  id: string;
  start: Date;
  end: Date;
  color: 'blue' | 'red' | 'green' | 'yellow';
};

interface BookingCalendarProps {
  bookings?: Booking[];
  onNavigate?: (date: Date) => void;
  initialDate?: Date;
}

export function BookingCalendar({
  bookings = [],
  onNavigate,
  initialDate = new Date(),
}: BookingCalendarProps) {
  const [currentDate, setCurrentDate] = useState(initialDate);
  const startOfCurrentWeek = startOfWeek(currentDate, { weekStartsOn: 1 });

  const weekDates = Array.from({ length: 7 }, (_, i) =>
    addDays(startOfCurrentWeek, i),
  );
  const hours = Array.from({ length: 24 }, (_, i) => i);

  const navigateWeek = (direction: 'prev' | 'next') => {
    const newDate = addDays(currentDate, direction === 'prev' ? -7 : 7);
    setCurrentDate(newDate);
    onNavigate?.(newDate);
  };

  const getBookingsForDate = (date: Date) => {
    return bookings.filter((booking) => isSameDay(date, booking.start));
  };

  return (
    <div className="relative mx-auto h-full w-full max-w-6xl overflow-hidden bg-stone-900 shadow-2xl ring-1 ring-stone-800">
      <BookingCalendarHeader
        onPrevious={() => navigateWeek('prev')}
        onNext={() => navigateWeek('next')}
        onToday={() => setCurrentDate(new Date())}
        weekDates={weekDates}
      />

      <div className="grid min-w-[800px] grid-cols-[auto_repeat(7,_minmax(120px,_1fr))] overflow-x-auto">
        <BookingCalendarHoursColumn />
        {weekDates.map((date) => (
          <BookingCalendarDateColumn
            date={date}
            bookings={getBookingsForDate(date)}
          />
        ))}
      </div>
    </div>
  );
}
