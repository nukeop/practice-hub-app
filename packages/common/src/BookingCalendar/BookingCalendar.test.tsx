import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { addHours } from 'date-fns';

import { BookingCalendar } from './BookingCalendar';

const mockDate = new Date('2024-01-15T00:00:00.000Z'); // A Monday

describe('BookingCalendar', () => {
  it('should always be UTC', () => {
    expect(new Date().getTimezoneOffset()).toBe(0);
  });

  it('renders correctly with default props', () => {
    const { container } = render(<BookingCalendar initialDate={mockDate} />);
    expect(container).toMatchSnapshot();
  });

  it('displays correct date range in header', () => {
    render(<BookingCalendar initialDate={mockDate} />);
    expect(screen.getByText(/15 — 21 Jan 2024/)).toBeInTheDocument();
  });

  it('displays bookings correctly', () => {
    const bookings = [
      {
        id: '1',
        start: addHours(mockDate, 2), // 2 AM on Monday
        end: addHours(mockDate, 4), // 4 AM on Monday
        color: 'blue' as const,
      },
    ];

    render(<BookingCalendar initialDate={mockDate} bookings={bookings} />);

    expect(screen.getByText('02:00 - 04:00')).toBeInTheDocument();
  });

  it('returns to today when clicking Today button', () => {
    render(<BookingCalendar initialDate={mockDate} />);

    const todayButton = screen.getByRole('button', { name: /today/i });
    userEvent.click(todayButton);

    expect(screen.getByText(/15 — 21 Jan 2024/)).toBeInTheDocument();
  });

  it('navigates to next week when clicking Next button', async () => {
    const onNavigate = jest.fn();
    render(<BookingCalendar initialDate={mockDate} onNavigate={onNavigate} />);

    const nextButton = screen.getByTestId('next-week');
    await userEvent.click(nextButton);

    expect(onNavigate).toHaveBeenCalledWith(addHours(mockDate, 7 * 24));
  });

  it('navigates to previous week when clicking Previous button', async () => {
    const onNavigate = jest.fn();
    render(<BookingCalendar initialDate={mockDate} onNavigate={onNavigate} />);

    const prevButton = screen.getByTestId('prev-week');
    await userEvent.click(prevButton);

    expect(onNavigate).toHaveBeenCalledWith(addHours(mockDate, -7 * 24));
  });

  it('displays all 24 hours', () => {
    render(<BookingCalendar initialDate={mockDate} />);
    expect(screen.getByText('00:00')).toBeInTheDocument();
    expect(screen.getByText('23:00')).toBeInTheDocument();
  });

  it('displays all weekdays', () => {
    render(<BookingCalendar initialDate={mockDate} />);
    ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'].forEach((day) => {
      expect(screen.getByText(day)).toBeInTheDocument();
    });
  });
});
