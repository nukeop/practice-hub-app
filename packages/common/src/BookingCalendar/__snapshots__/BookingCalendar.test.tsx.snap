// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCalendar renders correctly with default props 1`] = `
<div>
  <div
    class="relative mx-auto h-full w-full max-w-7xl overflow-hidden bg-stone-900 shadow-2xl ring-1 ring-stone-800"
  >
    <div
      class="flex items-center justify-between border-b border-stone-800 p-6"
    >
      <h2
        class="bg-gradient-to-b from-stone-200 to-stone-300 bg-clip-text text-lg font-bold text-transparent"
      >
        15
         â€” 
        21 Jan 2024
      </h2>
      <div
        class="flex gap-2"
      >
        <button
          class="rounded-full px-2 py-1 font-bold flex flex-row items-center justify-center transition-colors leading-none hover:bg-stone-800 ring-stone-800 text-stone-400 hover:ring-stone-700 ring-1 hover:text-stone-300 bg-transparent"
        >
          Today
        </button>
        <button
          class="rounded-full px-2 py-1 font-bold flex flex-row items-center justify-center transition-colors leading-none hover:bg-stone-800 ring-stone-800 text-stone-400 hover:ring-stone-700 ring-1 hover:text-stone-300 bg-transparent"
          data-testid="prev-week"
        >
          <svg
            class="lucide lucide-chevron-left h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m15 18-6-6 6-6"
            />
          </svg>
        </button>
        <button
          class="rounded-full px-2 py-1 font-bold flex flex-row items-center justify-center transition-colors leading-none hover:bg-stone-800 ring-stone-800 text-stone-400 hover:ring-stone-700 ring-1 hover:text-stone-300 bg-transparent"
          data-testid="next-week"
        >
          <svg
            class="lucide lucide-chevron-right h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="overflow-x-auto"
    >
      <div
        class="grid min-w-[800px] grid-cols-[auto_repeat(7,_minmax(120px,_1fr))]"
      >
        <div
          class="sticky left-0 z-10 bg-stone-900"
        >
          <div
            class="h-4 border-b border-stone-800"
          />
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            00:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            01:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            02:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            03:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            04:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            05:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            06:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            07:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            08:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            09:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            10:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            11:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            12:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            13:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            14:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            15:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            16:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            17:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            18:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            19:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            20:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            21:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            22:00
          </div>
          <div
            class="flex h-4 items-center border-b border-r border-stone-800 px-2 text-sm text-stone-400"
          >
            23:00
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Mon
              </div>
              <div
                class="text-base"
              >
                15
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Tue
              </div>
              <div
                class="text-base"
              >
                16
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Wed
              </div>
              <div
                class="text-base"
              >
                17
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Thu
              </div>
              <div
                class="text-base"
              >
                18
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Fri
              </div>
              <div
                class="text-base"
              >
                19
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Sat
              </div>
              <div
                class="text-base"
              >
                20
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-8 items-center justify-center border-b border-stone-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-base text-stone-400"
              >
                Sun
              </div>
              <div
                class="text-base"
              >
                21
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
            <div
              class="group relative h-4 border-b border-r border-stone-800"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
